src/Common/ProxyListConfigurationResolver.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <base/types.h>
+
+#include <Common/ProxyConfigurationResolver.h>
+#include <Poco/URI.h>
+
+namespace DB
+{
+
+/*
+ * Round-robin proxy list resolver.
+ * */
+class ProxyListConfigurationResolver : public ProxyConfigurationResolver
+{
+public:
+    explicit ProxyListConfigurationResolver(std::vector<Poco::URI> proxies_);
+
+    ProxyConfiguration resolve(bool https) override;
+
+    void errorReport(const ProxyConfiguration &) override {}
+
+private:
+    std::vector<Poco::URI> proxies;
+
+    /// Access counter to get proxy using round-robin strategy.
+    std::atomic<size_t> access_counter;
```

Comment:
Why is that atomic? Will the `ProxyListConfigurationResolver` instance be used from multiple threads?

Reply:
It was implemented like this, didn't want to change / break stuff. I'll investigate it later, but it's prolly safer to simply keep it.